src/spatial/explog.hpp

```diff
@@ -58,7 +58,7 @@ namespace se3
   template <typename D> Eigen::Matrix<typename D::Scalar,3,1,Eigen::internal::traits<D>::Options>
   log3(const Eigen::MatrixBase<D> & R)
   {
-    EIGEN_STATIC_ASSERT_MATRIX_SPECIFIC_SIZE(D, 3, 3);
+    EIGEN_STATIC_ASSERT_SAME_MATRIX_SIZE(D, Eigen::Matrix3d);
```

Comment:
`EIGEN_STATIC_ASSERT_MATRIX_SPECIFIC_SIZE` is strict while `EIGEN_STATIC_ASSERT_SAME_MATRIX_SIZE` is permissive. See http://eigen.tuxfamily.org/dox/TopicAssertions.html

`log3(R0.transpose() * R1)` was not compiling because of `EIGEN_STATIC_ASSERT_MATRIX_SPECIFIC_SIZE`. The work around is `log3((R0.transpose() * R1).eval())`.
With `EIGEN_STATIC_ASSERT_MATRIX_SPECIFIC_SIZE`, it compiles correctly.
I think this is a bug in some traits somewhere and it has been corrected lately.